src/platform/cert_capi_openssl.c

```diff
@@ -0,0 +1,266 @@
+            "[ lib] ERROR, %s.",
+            "d2i_X509 failed");
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
+        goto Exit;
+    }
+
+    KeyHandle = (NCRYPT_KEY_HANDLE)QuicCertGetPrivateKey(Cert);
+    if (KeyHandle == 0) {
+        Status = QUIC_STATUS_INTERNAL_ERROR;
+        goto Exit;
+    }
+
+    if (FAILED(Status =
+        NCryptGetProperty(
+            KeyHandle,
+            NCRYPT_EXPORT_POLICY_PROPERTY,
+            (PBYTE)&ExportPolicyProperty,
+            sizeof(ExportPolicyProperty),
+            &ExportPolicyLength, 0))) {
+        QuicTraceEvent(
+            LibraryError,
+            "[ lib] ERROR, %s.",
+            "NCryptGetProperty failed");
+        goto Exit;
+    }
+
+    if ((ExportPolicyProperty & NCRYPT_ALLOW_PLAINTEXT_EXPORT_FLAG) == 0) {
+        QuicTraceEvent(
+            LibraryError,
+            "[ lib] ERROR, %s.",
+            "Requested certificate does not support exporting. An exportable certificate is required");
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Exit;
+    }
+
+    if (FAILED(
+        Status =
+            NCryptExportKey(
+                KeyHandle,
+                0,
+                BCRYPT_RSAFULLPRIVATE_BLOB,
+                NULL,
+                NULL,
+                0,
+                &KeyLength,
+                0))) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "NCryptExportKey failed. Need exportable certificate");
+        goto Exit;
+    }
+
+    KeyData = QUIC_ALLOC_NONPAGED(KeyLength, QUIC_POOL_TLS_RSA);
+    if (KeyData == NULL) {
+        QuicTraceEvent(
+            AllocFailure,
+            "Allocation of '%s' failed. (%llu bytes)",
+            "RSA Key",
+            KeyLength);
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
+        goto Exit;
+    }
+
+    if (FAILED(Status =
+        NCryptExportKey(
+            KeyHandle,
+            0,
+            BCRYPT_RSAFULLPRIVATE_BLOB,
+            NULL,
+            KeyData,
+            KeyLength,
+            &KeyLength,
+            0))) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "NCryptExportKey failed. Need exportable certificate");
+        goto Exit;
+    }
+
+    BCRYPT_RSAKEY_BLOB* Blob = (BCRYPT_RSAKEY_BLOB*)KeyData;
+
+    if (Blob->Magic != BCRYPT_RSAFULLPRIVATE_MAGIC) {
+        QuicTraceEvent(
+            LibraryError,
+            "[ lib] ERROR, %s.",
+            "NCryptExportKey resulted in incorrect magic number");
+        Status = QUIC_STATUS_INTERNAL_ERROR;
+        goto Exit;
+    }
+
+    Rsa = RSA_new();
+    if (Rsa == NULL) {
+        QuicTraceEvent(
+            LibraryError,
+            "[ lib] ERROR, %s.",
+            "RSA_new failed");
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
+        goto Exit;
+    }
+
+    // n is the modulus common to both public and private key
```

Comment:
It is probably worth a block comment before all of this to explain how you came to this code, and generally what it is doing.